File: gerenciapet-backend/src/main/java/br/ufes/gerenciapet/backend/model/Tutor.java
package br.ufes.gerenciapet.backend.model;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

/**
 * Entidade que representa um tutor de grupo PET.
 */
@Entity
public class Tutor {

    /**
     * Identificador único do tutor.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Numero SIAPE do tutor.
     */
    @Column(nullable = false, length = 50)
    private String siape;

    /**
     * Departamento academico do tutor.
     */
    @Column(nullable = false, length = 100)
    private String departamento;

    /**
     * Grupo PET coordenado pelo tutor.
     */
    @OneToOne(mappedBy = "tutorCoordenador", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private GrupoPet grupoPetCoordena;

    /**
     * Usuário associado ao tutor.
     */
    @OneToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    /**
     * Retorna o identificador do tutor.
     *
     * @return identificador do tutor.
     */
    public Long getId() { return id; }
    /**
     * Define o identificador do tutor.
     *
     * @param id identificador do tutor.
     */
    public void setId(Long id) { this.id = id; }
    /**
     * Retorna o SIAPE do tutor.
     *
     * @return SIAPE do tutor.
     */
    public String getSiape() { return siape; }
    /**
     * Define o SIAPE do tutor.
     *
     * @param siape SIAPE do tutor.
     */
    public void setSiape(String siape) { this.siape = siape; }
    /**
     * Retorna o departamento do tutor.
     *
     * @return departamento do tutor.
     */
    public String getDepartamento() { return departamento; }
    /**
     * Define o departamento do tutor.
     *
     * @param departamento departamento do tutor.
     */
    public void setDepartamento(String departamento) { this.departamento = departamento; }
    /**
     * Retorna o grupo PET coordenado.
     *
     * @return grupo PET coordenado.
     */
    public GrupoPet getGrupoPetCoordena() { return grupoPetCoordena; }
    /**
     * Define o grupo PET coordenado.
     *
     * @param grupoPetCoordena grupo PET coordenado.
     */
    public void setGrupoPetCoordena(GrupoPet grupoPetCoordena) { this.grupoPetCoordena = grupoPetCoordena; }
    /**
     * Retorna o usuário associado ao tutor.
     *
     * @return usuário associado.
     */
    public User getUser() { return user; }
    /**
     * Define o usuário associado ao tutor.
     *
     * @param user usuário associado.
     */
    public void setUser(User user) { this.user = user; }
}

File: gerenciapet-frontend/src/app/models/Tutor.ts
import User from "./User";
import GrupoPet from "./GrupoPet";

export default class Tutor {
    public id: number = 0;
    public siape: string = '';
    public departamento: string = '';
    public grupoPetCoordena?: GrupoPet;
    public user?: User;
    public href: string = '';

    constructor(object?: any) {
        if (object !== undefined) {
            this.updateTutor(object);
        }
    }

    public updateTutor(object: any) {
        this.id = object.id ?? 0;
        this.siape = object.siape ?? '';
        this.departamento = object.departamento ?? '';
        if (object.grupoPetCoordena) {
            this.grupoPetCoordena = object.grupoPetCoordena;
        }
        if (object.user) {
            this.user = new User(object.user);
        }

        if (object._links !== undefined) {
            this.href = object._links.self?.href || this.href;
        } else {
            this.href = object.href ?? '';
        }
    }

    public getId(): string {
        if (this.id !== 0) return this.id.toString();
        if (this.href !== "") {
            let id = this.href.split("/").pop();
            if (id === undefined) return "0";
            else return id;
        }
        return "0";
    }
}
